### PR TITLE
caf: ble_adv: Remove BT_LE_ADV_OPT_USE_NAME

### PR DESCRIPTION
**Pull request by MarekPieta**
Change removes BT_LE_ADV_OPT_USE_NAME from advertising options and introduces a dedicated provider to keep whole scan response data configuration in one place.

Change also enables the Bluetooth device name scan response data provider in nRF Desktop and nRF Machine Learning applications. The device name is shortened if it does not fit in scan response data.

Jira: NCSDK-8848

### CODE REVIEW

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -192,7 +192,7 @@ nRF5340 Audio
 nRF Machine Learning (Edge Impulse)
 -----------------------------------
 
-|no_changes_yet_note|
+* Bluetooth device name is included in scan response data only if used Bluetooth local identity has no bond.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:195]:
```suggestion
* Updated the Bluetooth device name logic.
  The name is now included in scan response data only if the used Bluetooth local identity has no bond.
```
Please propagate to other occurrences.

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-changelog.rst (hunk 1 above)
Comment (MarekPieta) [reply doc/nrf/releases/release-notes-changelog.rst:195]: Changed to:
```
* Updated Bluetooth advertising data and scan response data logic.                                   
  The Bluetooth device name is included in scan response data only if used Bluetooth local identity has no bond.
```

Comment (MarekPieta) [reply doc/nrf/releases/release-notes-changelog.rst:195]: We actually updated logic related to Bluetooth advertising/scan response data, not logic related to Bluetooth device name itself.

Comment (MarekPieta) [reply doc/nrf/releases/release-notes-changelog.rst:195]: Later on we could merge this note with the doc changes introduced here: https://github.com/nrfconnect/sdk-nrf/pull/8457

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -192,7 +192,8 @@ nRF5340 Audio
 nRF Machine Learning (Edge Impulse)
 -----------------------------------
 
-|no_changes_yet_note|
+* Updated Bluetooth advertising data and scan response data logic.
+  The Bluetooth device name is now included in the scan response data only if used Bluetooth local identity has no bond.
```
Comment (divipillai) [inline doc/nrf/releases/release-notes-changelog.rst:196]: Shouldn't we add the use of "config BT_LE_ADV_PROV_DEVICE_NAME" Kconfig option also. I think that is now used to scan the response data.

Comment (MarekPieta) [reply doc/nrf/releases/release-notes-changelog.rst:196]: I think that this is more like an implementation detail. The generic information that we stated to use Bluetooth LE advertising data providers to configure the advertising data and scan response data would be included here: https://github.com/nrfconnect/sdk-nrf/pull/8457.
Can we leave it as it is?

Comment (divipillai) [reply doc/nrf/releases/release-notes-changelog.rst:196]: may be:
```suggestion
  The Bluetooth device name is now included in the scan response data, using the :ref:`CONFIG_BT_LE_ADV_PROV_FLAGS <config_desktop_app_options>` option, only if the used Bluetooth local identity has no bond.
```
Same applies for the other entries as well.
I am fine if the information about the scan response data is included as part of PR: https://github.com/nrfconnect/sdk-nrf/pull/8457.

Comment (MarekPieta) [reply doc/nrf/releases/release-notes-changelog.rst:196]: The `CONFIG_BT_LE_ADV_PROV_FLAGS` is an NCS option, so we should use :kconfig:option: syntax.

I am not sure if we need to go into details and mention Kconfig option names in the release notes (I think we could just inform in nRF Desktop documentation where the advertising data configuration is located). User would need to get familiar with new configuration method anyway. I try to keep the release notes simple and avoid going into details here.
If you insist on adding direct Kconfig references here, I could add them of course

Comment (divipillai) [reply doc/nrf/releases/release-notes-changelog.rst:196]: I also had an offline conversation with @greg-fer, and we could keep the content as is if we add the information to the nRF Desktop and other pages, whether this is relevant.

**Review by divipillai — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

**Comment by MarekPieta** [comment]
Hi @divipillai, @greg-fer
I needed to update the implementation to align with CI tests (doc had to be updated aswell). Please rereview.

**Review by greg-fer — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

**Review by zycz — APPROVED** [review]
Please add information about Bluetooth device name provider in documentation of advertising providers: doc/nrf/libraries/bluetooth_services/bt_le_adv_prov.rst

**Review by emob-nordic — APPROVED** [review]

**Review by kapi-no — APPROVED** [review]

**Review by KAGA164 — APPROVED** [review]

**Pull request merged**